### DISCUSSION TITLE
x25519 encrypt/decrypt?

### DISCUSSION THREAD

**Discussion by getify** [paulmillr/noble-curves#123] [Q&A]
I saw in another issue that you intentionally don't implement montgomery curve operations. I believe/assume this means you intentionally don't have an `encrypt()` / `decrypt()` method pair on the `x25519` API, correct?

Is there a way to do so, perhaps with the abstract API or utils, given having the x25519 keypair ready?

I'd like to do something like this:

```js
let encBuffer = encrypt(msgBuffer, x25519pk, ivBuffer);
let msgBuffer_2 = decrypt(encBuffer, x25519sk, ivBuffer);
```

Thanks for any guidance.

**Comment by paulmillr** [comment]
If you are talking about encrypting-to-public-key aka hybrid encryption, this is out of scope for noble-curves, because it involves using ciphers (noble-ciphers). The encryption itself does not involve curves at all. And we only provide low-level primitives per-package.

Dajiaji built HPKE.js on top of noble, conforming to RFC 9180, which has x25519 based encryption: https://github.com/dajiaji/hpke-js/blob/main/x/dhkem-x25519/README.md

**Comment by getify** [comment]
Sorry, I hadn't fully grasped the division of Curves (vs Hashes and Ciphers). I was thinking of them as a single package, and I just filed the discussion here. I guess this discussion belongs in the Ciphers repo, but the github UI isn't letting me transfer it.

I opened a new (more specific) discussion there: https://github.com/paulmillr/noble-ciphers/discussions/32 The discussion here can be closed I think.
